pydm/widgets/analog_indicator.py

```diff
@@ -0,0 +1,1576 @@
+
+    @Property(Qt.Edge)
+    def valuePosition(self):
+        """
+        The position of the value label (Top, Bottom, Left or Right).
+
+        Returns
+        -------
+        int
+            Qt.TopEdge, Qt.BottomEdge, Qt.LeftEdge or Qt.RightEdge
+        """
+        return self._value_position
+
+    @valuePosition.setter
+    def valuePosition(self, position):
+        """
+       The position of the value label (Top, Bottom, Left or Right).
+
+        Parameters
+        ----------
+        position : int
+            Qt.TopEdge, Qt.BottomEdge, Qt.LeftEdge or Qt.RightEdge
+        """
+        self._value_position = position
+        self.setup_widgets_for_orientation(self.orientation, self.flipScale, self.invertedAppearance, position)
+
+    @Property(bool)
+    def originAtZero(self):
+        """
+        Whether or not the scale indicator should start at zero value.
+        Applies only for bar indicator.
+
+        Returns
+        -------
+        bool
+        """
+        return self.scale_indicator.get_origin_at_zero()
+
+    @originAtZero.setter
+    def originAtZero(self, checked):
+        """
+        Whether or not the scale indicator should start at zero value.
+        Applies only for bar indicator.
+
+        Parameters
+        ----------
+        checked : bool
+        """
+        self.scale_indicator.set_origin_at_zero(checked)
+
+    @Property(bool)
+    def limitsFromChannel(self):
+        """
+        Whether or not the scale indicator should use the limits information
+        from the channel.
+
+        Returns
+        -------
+        bool
+        """
+        return self._limits_from_channel
+
+    @limitsFromChannel.setter
+    def limitsFromChannel(self, checked):
+        """
+        Whether or not the scale indicator should use the limits information
+        from the channel.
+
+        Parameters
+        ----------
+        checked : bool
+            True to use the limits from the Channel, False to use the user-defined
+            values.
+        """
+        print("_limits_from_channel")
+        print(self._limits_from_channel)
+        print("checked outside of if (limits)")
+        print(checked)
+        print("_lower_ctrl_limit")
+        print(self._lower_ctrl_limit)
+        if self._limits_from_channel != checked:
+            print("inside limists if")
+            self._limits_from_channel = checked
+            if checked:
+                if self._lower_ctrl_limit:
+                    self.scale_indicator.set_lower_limit(self._lower_ctrl_limit)
+                if self._upper_ctrl_limit:
+                    self.scale_indicator.set_upper_limit(self._upper_ctrl_limit)
+            else:
+                self.scale_indicator.set_lower_limit(self._user_lower_limit)
+                self.scale_indicator.set_upper_limit(self._user_upper_limit)
+            self.update_labels()
+
+
+
+    @Property(float)
+    def userUpperLimit(self):
```

Comment:
Minor designer usability thing: since all of the "upper" limits are defined before the "lower" limits, they appear in the order of upper first, lower second in the designer editor. So if I want a range from 30 to 50, for example, my designer editor looks like:
```
userUpperLimit         50
userLowerLimit         30
```
which reads out in my brain as "from 50 to 30" and breaks me a little. I think the other order would be more intuitive so I could read down "from low to high" in the settings.